test/integration/scheduler/extender_test.go

```diff
@@ -61,45 +63,69 @@ type priorityConfig struct {
 }
 
 type Extender struct {
-	name         string
-	predicates   []fitPredicate
-	prioritizers []priorityConfig
+	name          string
+	predicates    []fitPredicate
+	prioritizers  []priorityConfig
+	filteredNodes []v1.Node
 }
 
 func (e *Extender) serveHTTP(t *testing.T, w http.ResponseWriter, req *http.Request) {
-	var args schedulerapi.ExtenderArgs
+	if strings.Contains(req.URL.Path, filter) || strings.Contains(req.URL.Path, prioritize) {
+		var args schedulerapi.ExtenderArgs
 
-	decoder := json.NewDecoder(req.Body)
-	defer req.Body.Close()
+		decoder := json.NewDecoder(req.Body)
+		defer req.Body.Close()
 
-	if err := decoder.Decode(&args); err != nil {
-		http.Error(w, "Decode error", http.StatusBadRequest)
-		return
-	}
+		if err := decoder.Decode(&args); err != nil {
+			http.Error(w, "Decode error", http.StatusBadRequest)
+			return
+		}
 
-	encoder := json.NewEncoder(w)
+		encoder := json.NewEncoder(w)
 
-	if strings.Contains(req.URL.Path, filter) {
-		resp := &schedulerapi.ExtenderFilterResult{}
-		nodes, failedNodes, err := e.Filter(&args.Pod, &args.Nodes)
-		if err != nil {
-			resp.Error = err.Error()
-		} else {
-			resp.Nodes = *nodes
-			resp.FailedNodes = failedNodes
+		if strings.Contains(req.URL.Path, filter) {
+			resp := &schedulerapi.ExtenderFilterResult{}
+			nodes, failedNodes, err := e.Filter(&args.Pod, &args.Nodes)
+			if err != nil {
+				resp.Error = err.Error()
+			} else {
+				resp.Nodes = *nodes
+				resp.FailedNodes = failedNodes
+			}
+
+			if err := encoder.Encode(resp); err != nil {
+				t.Fatalf("Failed to encode %+v", resp)
+			}
+		} else if strings.Contains(req.URL.Path, prioritize) {
+			// Prioritize errors are ignored. Default k8s priorities or another extender's
+			// priorities may be applied.
+			priorities, _ := e.Prioritize(&args.Pod, &args.Nodes)
+
+			if err := encoder.Encode(priorities); err != nil {
+				t.Fatalf("Failed to encode %+v", priorities)
+			}
+		}
+	} else if strings.Contains(req.URL.Path, bind) {
+		var args schedulerapi.Binding
+
+		decoder := json.NewDecoder(req.Body)
```

Comment:
More of a nit, but you have repeating blocks that can be pulled out above the 1st if.